lib/msal-browser/src/utils/BrowserUtils.ts

```diff
@@ -20,12 +21,17 @@ export class BrowserUtils {
      * @param {string} urlNavigate - URL of the authorization endpoint
      * @param {boolean} noHistory - boolean flag, uses .replace() instead of .assign() if true
      */
-    static navigateWindow(urlNavigate: string, noHistory?: boolean): void {
+    static navigateWindow(urlNavigate: string, logger: Logger, noHistory?: boolean): void {
         if (noHistory) {
             window.location.replace(urlNavigate);
         } else {
             window.location.assign(urlNavigate);
         }
+
+        // To block code from running after navigation, this should not throw if navigation succeeds
+        setTimeout(() => {
```

Comment:
This won't block the code from executing, it just makes it so a log message gets printed out. I like the idea of making this function return a promise, and resolving that promise this time.

Reply:
Required a little bit of refactor of the `RedirectHandler` but implemented your suggestion. Thanks!